oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsPowerSource.java

```diff
@@ -152,8 +152,10 @@ private static WindowsPowerSource getPowerSource(String name) {
         HANDLE hdev = SetupApi.INSTANCE.SetupDiGetClassDevs(GUID_DEVCLASS_BATTERY, null, null,
                 SetupApi.DIGCF_PRESENT | SetupApi.DIGCF_DEVICEINTERFACE);
         if (WinBase.INVALID_HANDLE_VALUE != hdev) {
+        	boolean batteryFound = false;
             // Limit search to 100 batteries max
             for (int idev = 0; idev < 100; idev++) {
+            	if (batteryFound == false) {
```

Comment:
This would work, but there are two improvements you can make.
1. Checking that a boolean `==` true or false is easier done just evaluating its value, so `batteryFound` is either true or false by itself.  `!batteryFound` will return true if the value is false, which is what we want to check.
2. You can include this check in the iteration check for the for loop right above it:
`for (int idev = 0; !batteryFound && idev < 100; idev++)`

Reply:
So it's efficient to remove that 'if' statement wholly and instead incorporate the boolean check in the 'for' loop itself?

Reply:
Sir, I just did the changes locally and ran the Maven test. It passed. What should I do next? Should I submit another pull request? I haven't worked in this code review environment before, so I don't really know how to proceed.
Thanks.

Reply:
Just commit the new change to your local branch and "push" it to your github account. It will automatically update the pull request here.

Reply:
As for "efficiency" it's probably not measurable. On the last iteration it would be slightly more efficient to check the "not found" before the iteration limit, but given branch prediction it likely will make zero practical difference.  And the compiler would optimize out the `== false` check as well.   It's more just readability and concise code! :)

Reply:
Okay sir! I'll do it right now!